language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs

```diff
@@ -193,55 +174,65 @@ tests damlTypes yarn damlc daml2ts davl = testGroup "daml2ts tests"
+        daml2tsProject [groverDar, groverDar] daml2tsDir
         assertFileExists (groverTsSrc </> "Grover.ts")
+        assertFileExists (groverTsLib </> "Grover.js")
+        assertFileExists (groverTsLib </> "Grover.d.ts")
         assertFileExists (groverTsSrc </> "packageId.ts")
+        assertFileExists (groverTsLib </> "packageId.js")
+        assertFileExists (groverTsLib </> "packageId.d.ts")
 
   , testCaseSteps "DAVL test" $ \step -> withTempDir $ \here -> do
       let daml2tsDir = here </> "daml2ts"
       withCurrentDirectory here $ do
+        copyDirectory damlTypes "daml-types"
         step "daml2ts..."
-        -- Call daml2ts once without a 'package.json'.
+        writePackageJson
         callProcessSilent daml2ts $
           [ davl </> "davl-v4.dar"
           , davl </> "davl-v5.dar"
           , davl </> "davl-upgrade-v4-v5.dar" ] ++
           ["-o", daml2tsDir]
-        assertFileExists (here </> "package.json")
-        -- Overwrite the 'package.json' that daml2ts generated because
-        -- we need to adjust module resolution for @daml/types and
-        -- @daml/ledger.
-        setupWorkspace
-        -- Call daml2ts again which will this time update 'package.json'.
-        callProcessSilent daml2ts $
-          [ davl </> "davl-v4.dar"
-          , davl </> "davl-v5.dar"
-          , davl </> "davl-upgrade-v4-v5.dar" ] ++
-          ["-o", daml2tsDir] -- There's no need to pass '-p
-                             -- here/package.json' but we could and it would mean the same.
         assertFileExists (daml2tsDir </> "davl-0.0.4" </> "src" </> "DAVL.ts")
-        assertFileExists (daml2tsDir </> "davl-0.0.5" </> "src" </> "DAVL.ts")
-        assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "src" </> "Upgrade.ts")
-        step "yarn install..."
-        yarnProject ["install"]
-        step "yarn workspaces run build..."
-        yarnProject ["workspaces", "run", "build"]
         assertFileExists (daml2tsDir </> "davl-0.0.4" </> "lib" </> "DAVL.js")
+        assertFileExists (daml2tsDir </> "davl-0.0.4" </> "lib" </> "DAVL.d.ts")
+        assertFileExists (daml2tsDir </> "davl-0.0.5" </> "src" </> "DAVL.ts")
         assertFileExists (daml2tsDir </> "davl-0.0.5" </> "lib" </> "DAVL.js")
+        assertFileExists (daml2tsDir </> "davl-0.0.5" </> "lib" </> "DAVL.d.ts")
+        assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "src" </> "Upgrade.ts")
         assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "lib" </> "Upgrade.js")
-        step "yarn workspaces run lint..."
-        yarnProject ["workspaces", "run", "lint"]
-     ]
+        assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "lib" </> "Upgrade.d.ts")
+      step "eslint..."
+      withCurrentDirectory daml2tsDir $ do
+        pkgs <- (\\ ["package.json", "node_modules"]) <$> listDirectory daml2tsDir
+        BSL.writeFile "package.json" $ encode (
```

Comment:
We should probably lift these constants to the type of the file for easy maintenance as we do in `TsCodeGenMain.hs`.